hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleAppTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.console;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+/**
+ * Tests for demo client console application.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
+public class ClientConsoleAppTest extends HazelcastTestSupport {
+
+    private static final PrintStream systemOutOrig = System.out;
+
+    private static ByteArrayOutputStream baos;
+
+    private  HazelcastInstance hazelcastInstance;
+
+    @BeforeClass
+    public static void beforeClass() {
+        baos = new ByteArrayOutputStream();
+        try {
+            System.setOut(new PrintStream(baos, true, "UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            // Should never happen for the UTF-8
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        System.setOut(systemOutOrig);
```

Comment:
Will this set the stdout for all other tests that may be running in parallel in the same JVM?

Reply:
yes - that is global property, i assumed that it is ok to use that approach because ConsoleAppTest also do it ( i had doubts but :D )

Reply:
I would suggest redesign so that it is not globally set not to affect other tests. I see that it is used by `ConsoleAppTest` as well but I think it is not a good design, do you have a better alternative design for testing?

Reply:
I will play around that at home.

Reply:
@ihsandemir fixed (at least i think so) - I am passing output buffer explicitly, no more sharing global state.